# PyBank Homework week 3
* my tasks for this assignment are to:
    * Calculate the total months included in the dataset
    * The net total amount of Profit/lossses over the entire period
    * The average of the changes in Profit/Losses over the entire period
    * The greatest increase in profits(date and amount) over the entire period
    * The greatest decrease in losses( date and amount) over the entire period
    

#### First I need to import the dataset into a readable file for python

In [9]:
import pandas as pd
ramen_sales=pd.read_csv('budget_data.csv')

#### review the files to make sure it imported correctly 

In [10]:
ramen_sales.head()

Date  Profit/Losses
0  Jan-2010         867884
1  Feb-2010         984655
2  Mar-2010         322013
3  Apr-2010         -69417
4  May-2010         310503

In [ ]:
ramen_sales.set_index('Date', inplace=True)

In [112]:
ramen_sales

Profit/Losses
Date                   
Jan-2010         867884
Feb-2010         984655
Mar-2010         322013
Apr-2010         -69417
May-2010         310503
...                 ...
Oct-2016         102685
Nov-2016         795914
Dec-2016          60988
Jan-2017         138230
Feb-2017         671099

[86 rows x 1 columns]

* now cleaner to sort through

#### Finding the total number of months included 

In [113]:
ramen_sales.info()

<class 'pandas.core.frame.DataFrame'>
Index: 86 entries, Jan-2010 to Feb-2017
Data columns (total 1 columns):
 #   Column         Non-Null Count  Dtype
---  ------         --------------  -----
 0   Profit/Losses  86 non-null     int64
dtypes: int64(1)
memory usage: 1.3+ KB


* or we can find this information

In [114]:
ramen_sales.shape

(86, 1)

* we see that there are *86* rows that are associated with months in date file

#### Finding the net total amount of profit/loss over the period

In [115]:
total_sales=int(ramen_sales.sum(numeric_only=True))
print(total_sales)

38382578


* this gives us a net total profit of $38,382,578

#### finding average of the changes

In [116]:
int(ramen_sales.sum(numeric_only=True))

38382578

In [117]:
average_change=(ramen_sales.max(numeric_only=True)-ramen_sales.min(numeric_only=True))/(86-1)
print(float(round(average_change, ndigits=2)))

27844.92


* average change in revenue is 27844

* formula for average change found here:
https://courses.lumenlearning.com/ivytech-collegealgebra/chapter/find-the-average-rate-of-change-of-a-function/#:~:text=The%20units%20on%20a%20rate,change%20in%20the%20input%20values.

#### finding the greatest increase in profits 

In [122]:
ramen_sales.max()

Profit/Losses    1170593
dtype: int64

In [190]:
max_date=ramen_sales['Profit/Losses'].idxmax()
print(f'Greatest Increase in Profit: {max_date} ${int(ramen_sales.max())}')

Greatest Increase in Profit: Feb-2012 $1170593


#### Greatest Decrease in Profit

In [194]:
min_date=ramen_sales['Profit/Losses'].idxmin()
print(f'Greatest Decrease in Profits; {min_date} ${int(ramen_sales.min())}')


Greatest Decrease in Profits; Sep-2013 $-1196225


#### Setting up the return Board

In [200]:
financial_analysis=print(f"""
Financial Analysis
------------------------
Total Months:86
Total: ${total_sales}
Average Change: ${float(round(average_change, ndigits=2))}
Greatest Increase in Profit: {max_date} ${int(ramen_sales.max())}
Greatest Decrease in Profit: {min_date} ${int(ramen_sales.min())}
""")


Financial Analysis
------------------------
Total Months:86
Total: $38382578
Average Change: $27844.92
Greatest Increase in Profit: Feb-2012 $1170593
Greatest Decrease in Profit: Sep-2013 $-1196225



#### extra while playing around with describle

In [21]:
ramen_sales.describe().apply(lambda s: s.apply('{0:f}'.format))

Profit/Losses
count        86.000000
mean     446309.046512
std      536357.947941
min    -1196225.000000
25%      182162.000000
50%      570328.000000
75%      795226.250000
max     1170593.000000

* 
    * formating for scientic notation found here:
    https://stackoverflow.com/questions/40347689/dataframe-describe-suppress-scientific-notation